### DISCUSSION TITLE
Basic project manager plugin

### DISCUSSION THREAD

**Discussion by j0lol** [nvim-mini/mini.nvim#934] [Ideas]
- like [coffeebar/neovim-project](https://github.com/coffebar/neovim-project) or [ahmedkhalf/project.nvim](https://github.com/ahmedkhalf/project.nvim)
- with mini.starter support maybe?

**Comment by j0lol** [comment]
if anyone is looking for a mini.starter integration, i wrote a function that gets recent projects from coffeebar/neovim-project :)

(edited with suggestions below)
```lua
local gen_proj_list = function(size)
    local projects = require("neovim-project.utils.history").get_recent_projects()
    local size = math.min(size, #projects)
    local starter_items = {}
    for i = 1, size do
        local project = projects[i]
        table.insert(starter_items, 1, { { name = vim.fs.basename(project), action = function()
            require("neovim-project.project").switch_project(project)
        end, section="Recent Projects" } })
    end
    return starter_items
end
```

**Reply by echasnovski** [reply]
Thanks for sharing! This indeed might be useful for future users.

I'd suggest precomputing final size with `size = math.min(size, #projects)` and then use `for i = 1, size do` loop instead. This allows to avoid `if i > size then` check.

**Reply by j0lol** [reply]
Thanks! I'm not very fluent in lua, so I appreciate the help :)

**Comment by echasnovski** [comment]
Thanks for the suggestion!

['mini.sessions'](https://github.com/echasnovski/mini.nvim/blob/main/readmes/mini-sessions.md) is intended as "project management" solution from 'mini.nvim'. At least that's how I use it and it seems to work fine. Especially together with 'mini.starter'.

**Reply by j0lol** [reply]
Thanks! I'll look into mini.sessions! I didn't look into it because it doesn't look like the typical IDE "projects", so I assumed it was something different or not applicable to my use case. Thanks for making mini.nvim!
